src/zsf_config.c

```diff
@@ -0,0 +1,81 @@
+
+#include "zsf_config.h"
+#include "ini.h"
+#include "csv.h"
+
+#include <assert.h>
+#include <string.h>
+#include <stdlib.h>
+#include <errno.h>
+
+int _zsf_ini_handler(char *section, char *key, char *value, void* data_ptr)
+{
+  zsf_config_t *config_ptr = (zsf_config_t*)data_ptr;
+  sealock_index_t lock_index = 0;
+  char *end_ptr = NULL;
+
+  assert(section);
+  assert(key);
+  assert(value);
+  assert(data_ptr);
+
+  errno = 0;
+  if (!strcmp(section, "sealock")) {
+    config_ptr->num_locks = 1;
+    if (!strcmp(key, "computation_mode")) {
+      if (!strcmp(value, "cycle_average")) {
+        config_ptr->locks[lock_index].computation_mode = cycle_average_mode;
+      } else if (!strcmp(value, "phase_wise")) {
+        config_ptr->locks[lock_index].computation_mode = phase_wise_mode;
+      } else {
+        return INI_FAIL;
+      }
+    } else if (!strcmp(key, "sealock_operational_parameters")) {
+      config_ptr->locks[lock_index].operational_parameters_file = strdup(value);
```

Comment:
strdup is technically C23 I guess? Perhaps that can be the standard we compile with

Reply:
Added C23 as the standard in cmake files.